napari/layers/_shapes_layer/shape_list.py

```diff
@@ -657,3 +657,36 @@ def to_labels(self, labels_shape=None, shape_type=None):
                     labels[mask] = index[ind]
 
         return labels
+
+    def to_xml(self, shape_type=None):
+        """Returns a list of xml elements defining each shape according to the
+        svg specification. Z ordering of the shapes will be taken into account.
+        Passing a `shape_type` argument leads to only xml from that particular
+        `shape_type` being returned.
+
+        Parameters
+        ----------
+        shape_type : {'line', 'rectangle', 'ellipse', 'path', 'polygon'} |
+                     None, optional
+            String of shape type to be included.
+
+        Returns
+        ----------
+        xml : list
+            List of xml elements defining each shape according to the
+            svg specification
+        """
+
+        if shape_type is None:
+            xml = [self.shapes[ind].to_xml() for ind in self._z_order[::-1]]
+        elif shape_type not in self._types.keys():
+            raise ValueError("""shape_type not recognized, must be one of
+                         "{'line', 'rectangle', 'ellipse', 'path',
+                         'polygon'}"
+                         """)
+        else:
+            cls = self._types[shape_type]
+            xml = [self.shapes[ind].to_xml() for ind in self._z_order[::-1] if
```

Comment:
```suggestion
            xml = [self.shapes[ind].to_xml() for ind in self._z_order.reverse() if
```

Reply:
fixed

Reply:
actually `_z_order` is a numpy array so sticking with `[::-1]`